### PR TITLE
Convert kustomize build docker run tests and remove NODOCKER option

### PR DESCRIPTION
**Pull request by prachirp**
Fixes https://github.com/GoogleContainerTools/kpt/issues/976

### CODE REVIEW

**Comment by prachirp** [comment]
New e2e tests depend on unreleased kpt features, will pass after next kpt release

File: tests/kustomize_build.sh [hunk 1]
```diff
@@ -21,20 +21,6 @@ DIR="$(dirname "$0")"
...
-
-# TODO: Revisit after adressing https://github.com/GoogleContainerTools/kpt/issues/839
-testcase "docker_kustomize_build_imperative_git"
-docker run -u "$(id -u)" gcr.io/kpt-functions/kustomize-build:"${TAG}" -d path=https://github.com/kubernetes-sigs/kustomize/examples/multibases/ |
-  docker run -i -u "$(id -u)" -v "$(pwd)":/sink gcr.io/kpt-functions/write-yaml:"${TAG}" -o /dev/null -d sink_dir=/sink -d overwrite=true
-assert_contains_string pod_cluster-a-staging-myapp-pod.yaml "name: cluster-a-staging-myapp-pod"
-
 ############################
 # kpt fn Tests
 ############################
```
Comment (frankfarzan) [inline tests/kustomize_build.sh:26]: Remove these headers as they're redundant. In other files too.

File: tests/kustomize_build.sh [hunk 2]
```diff
@@ -79,3 +65,26 @@ kpt fn source examples |
   kpt fn sink .
 assert_contains_string configmap_the-map.yaml "app: hello"
 assert_dir_exists knative-serving
+
+testcase "kpt_kustomize_build_declarative_git"
```
Comment (frankfarzan) [inline tests/kustomize_build.sh:69]: nit: for clarity, highlight the test cases as such:

############################
testcasse blah
############################

File: tests/kustomize_build.sh (hunk 1 above)
Comment (prachirp) [reply tests/kustomize_build.sh]: Done

File: tests/kustomize_build.sh (hunk 2 above)
Comment (prachirp) [reply tests/kustomize_build.sh:65]: Having the testcase name as the first line is clear enough.

File: tests/common.sh [hunk 3]
```diff
@@ -19,7 +19,6 @@
 set -eo pipefail
 
 TAG=${TAG:-dev}
-NODOCKER=${NODOCKER:-}
```
Comment (frankfarzan) [inline tests/common.sh:22]: can you also include the new 'sops' function in this PR?

**Review by frankfarzan — CHANGES_REQUESTED** [review]
Can you make sure cI passes?

File: tests/common.sh (hunk 3 above)
Comment (prachirp) [reply tests/common.sh:22]: Done

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**